core/src/main/java/io/questdb/griffin/engine/functions/regex/AbstractLikeStrFunctionFactory.java

```diff
@@ -54,7 +54,16 @@ public static String escapeSpecialChars(CharSequence pattern, CharSequence prev)
         StringSink sink = Misc.getThreadLocalBuilder();
         for (int i = 0; i < len; i++) {
             char c = pattern.charAt(i);
-            if (c == '_')
```

Comment:
() around "c == '\\' && i + 1 < len" are not necessary . 
If c == '\\' then it won't match the next 3 if conditions so it'd be good to either : 
- move whole condition just before the last else 
- handle second character not in "_%\\" in nested  if/else .